tcl/org.testeditor.tcl.dsl/src/org/testeditor/tcl/dsl/formatting2/TclFormatter.xtend

```diff
@@ -88,10 +89,16 @@ class TclFormatter extends XbaseFormatter {
 
 	def dispatch void format(SpecificationStepImplementation step, extension IFormattableDocument document) {
 		step.regionFor.keyword("*").prepend[newLines = 2]
-		step.contents.forEach[format]
 		step.regionFor.keyword(".").prepend[noSpace]
-
-		step.regionFor.feature(SPECIFICATION_STEP_IMPLEMENTATION__TEST).append[newLine]
+		val nlregion=step.regionFor.feature(SPECIFICATION_STEP_IMPLEMENTATION__NL)
+		addReplacer(new AbstractTextReplacer(document,nlregion) {
+			
+			override createReplacements(ITextReplacerContext context) {
+				context.addReplacement(region.replaceWith(region.text.replaceAll('(\r?\n)+(\t| )+','')))
```

Comment:
If I got this right, this forces specification steps onto a single line. But shouldn't the formatter retain a single space? Also, what if a have a line break but no tab or space following that. In that case, the line break remains – is that intentional?

Reply:
This is very dirty indeed and was the only way I could make the formatter work as expected. I believe the formatter misbehaves because of explicit whitespaces and newlines I introduced as separate terminals. I'm not happy with this solution and I should look into this again. Maybe I find a way to get rid of this dirty workaround ...

Reply:
Did not find a work around, I really tried. Since this is the formatter we'll have to postpone this for a better version of xtext which we have to upgrade to anyway. We should revisit this then ...